security/fma-interop-portal.md

```diff
@@ -0,0 +1,212 @@
+these functions are in scope for this contest.
+
+## OptimismPortal proving function for Super Roots
+
+The `OptimismPortal` is updated to have a new version of `proveWithdrawalTransaction` that uses the
+Super Root proving path. The Super Root proving path is toggled by the variable `superRootsActive`
+which is meant to be triggered when a chain migrates to using Super Roots.
+
+## Failure Modes
+
+### FM1: Incorrect State Reporting by `AnchorStateRegistry`
+
+#### Description
+
+A bug within the `AnchorStateRegistry` causes it to return incorrect data about a dispute game
+(e.g., resolves incorrectly, reports wrong `respectedGameType`, incorrect `retirementTimestamp`).
+Since the `OptimismPortal` now relies entirely on the `AnchorStateRegistry` as the source of truth
+for game validity and state, incorrect data leads directly to `OptimismPortal` malfunction.
+
+#### Risk Assessment
+
+- Impact: CRITICAL
+  - Reasoning: Could lead to a **Withdrawal Safety Failure** or **Withdrawal Liveness Failure**.
+- Likelihood: LOW
+  - Reasoning: Assumes the `AnchorStateRegistry` is well-tested and audited as a core component.
+
+#### Mitigations
+
+- Rigorous testing (unit, integration, fuzzing) of `AnchorStateRegistry`.
+- Multiple independent audits of `AnchorStateRegistry`.
+
+#### Detection
+
+- Monitoring already exists for the correctness and age of the Anchor State.
+- Monitoring already exists for the correctness of resolved dispute games.
+
+#### Recovery Path(s)
+
+- Depends heavily on the nature of the bug.
+- If possible, upgrade the `AnchorStateRegistry` contract to fix the bug.
+- May require manual intervention via governance or admin roles to correct state or pause affected
+  portal functions temporarily.
+- Possibly not recoverable if Withdrawal Safety is violated.
+
+### FM2: Bug in `ETHLockbox` / `OptimismPortal` Migration Logic
+
+#### Description
+
+The process for migrating ETH funds from an old `ETHLockbox` to a new one, coordinated by the
+`OptimismPortal` and `ETHLockbox`, involves complex logic. A bug in this logic could leave the
+system in a state that impacts either safety or liveness.
+
+#### Risk Assessment
+
+- Impact: HIGH/CRITICAL
+  - Reasoning: Failure could result in funds being partially transferred, locked indefinitely in
+    either contract, or lead to other inconsistencies between the Portal and the Lockbox,
+    potentially causing either a **Loss of Funds** or a **Withdrawal Liveness Failure**.
+- Likelihood: LOW
+  - Reasoning: Cross-contract interactions for state transitions like migration can be complex, but
+    the migration process is well-tested.
+
+#### Mitigations
+
+- Rigorous testing focused on migration scenarios, reusing existing upgrade testing.
+- Clear documentation of the expected state after the migration integrated into the superchain-ops
+  task that actually executes this migration.
+
+#### Recovery Path(s)
```

Comment:
Action items, as part of betanet:
- Do a withdrawal and see that it works
- Check lockbox balances make sense
- Check that withdrawal proofs are invalidated

Reply:
@smartcontracts: follow up with Proofs to understand what they are solving for and what requirements are not met by a trusted input.